### ISSUE TITLE
Where can I report vulnerability details?

### ISSUE THREAD

**Issue by orangetw** [koajs/koa#1128]
Hi, I found some vulns, Is there any security related mail I can send my report?

**Comment by jonathanong** [comment]
u can email me at me@jongleberry.com

**Comment by jonathanong** [comment]
i'll add my email in the docs for reporting vulnerabilities

**Comment by orangetw** [comment]
Hi, I have sent the vulnerability detail, please check it!

**Comment by maticrivo** [comment]
@orangetw @jonathanong it would be very interesting for everyone to know what exactly is the vulnerability about and what can get compromised and if we as koa users can do anything to protect ourselves..

**Comment by dotnil** [comment]
@maticrivo I believe reporting to maintainers in advance is the right approach. Further details may be revealed after there's a fix.

**Comment by maticrivo** [comment]
@dotnil of course, I'm saying after they review it and fix/patch the vulnerability

**Comment by maticrivo** [comment]
any update you can share about this?

**Comment by jonathanong** [comment]
i've received two reports. one is not about koa itself but one of its middleware (which is this issue). another is about setting headers, which was vague because they didn't describe the actual attack. yeah, a client can set whatever headers they want...

**Comment by maticrivo** [comment]
thanks for the update @jonathanong

**Comment by fl0w** [comment]
> yeah, a client can set whatever headers they want...

@jonathanong I can't seem to find the comment, but vaguely recall semi-consensus around adding some form of `app.trustHeaderfield()` API?

**Comment by pepkin88** [comment]
I would also add a possibility of spoofing an IP address when behind an Nginx server: https://github.com/koajs/koa/issues/599#issuecomment-239493311
